restart_process/README.md

```diff
@@ -1,10 +1,123 @@
+## How to Use
+
+Import this extension by putting the following at the top of your Tiltfile:
+```python
+load('ext://restart_process', 'docker_build_with_restart')
+```
+
+For the image that needs the process restart, replace your existing `docker_build` call:
+```python
+docker_build(
+    'foo-image',
+    './foo',
+    arg1=val1,
+    arg2=val2,
+    live_update=[x, y, z...]
+)
+```
+with a `docker_build_with_restart` call:
+```python
+docker_build_with_restart(
+    'foo-image',
+    './foo',
+    entrypoint='/go/bin/foo',
+    arg1=val1,
+    arg2=val2,
+    live_update=[x, y, z...]
+)
+```
+The call above looks just like the initial `docker_build` call except for one added parameter, `entrypoint` (in this example, `/go/bin/foo`). This is the command that you want to run on container start and _re-run_ on Live Update.
+
+### Function Signature
+```python
+def docker_build_with_restart(ref: str, context: str,
+    entrypoint: Union[str, List[str]],
+    live_update: List[LiveUpdateStep],
+    base_suffix: str = '-base',
+    restart_file: str = '/.restart-proc',
+    **kwargs
+):
+    """Args:
+      ref: name for this image (e.g. 'myproj/backend' or 'myregistry/myproj/backend'); as the parameter of the same name in docker_build
+      context: path to use as the Docker build context; as the parameter of the same name in docker_build
+      entrypoint: the command to be (re-)executed when the container starts or when a live_update is run
+      live_update: set of steps for updating a running container; as the parameter of the same name in docker_build
+      base_suffix: suffix for naming the base image, applied as {ref}{base_suffix}
+      restart_file: file that Tilt will update during a live_update to signal the entrypoint to rerun
+      **kwargs: will be passed to the underlying `docker_build` call
+    """
+```
+## Unsupported Cases
+This extension does NOT support process restarts for:
+- Images run in Docker Compose resources
+- `custom_build`
+
+Run into a bug? Need a use case that we don't yet support? Let us know--[open an issue](https://github.com/windmilleng/tilt-extensions/issues) or [contact us](https://tilt.dev/contact).
+
+## What's Happening Under the Hood
+*If you're a casual user/just want to get your app running, you can stop reading now. However, if you want to dig deep and know exactly what's going on, or are trying to debug weird behavior, read on.*
+
+This extension wraps commands in `tilt-restart-wrapper`, which makes use of [`entr`](https://github.com/eradman/entr/)
+to run arbitrary commands whenever a specified file changes. Specifically, we override the container's entrypoint with the following:
+
+```
+/tilt-restart-wrapper --watchfile='/.restart-proc' <entrypoint>
+```
+
+This invocation says:
+- when the container starts, run <entrypoint>
+- whenever the `/.restart-proc` file changes, re-execute <entrypoint>
+
+We also set the following as the last `live_update` step:
+```python
+run('date > /.restart-proc')
```

Comment:
huh, it doesn't have touch, but it has date? wild.

Reply:
No, I didn't even test this part of the flow on `scratch` because it failed earlier in the live update (at the sync step). I use `date` instead of `touch` because entr doesn't respond to timestamp changes, only to file writes (see [entr#32](https://github.com/eradman/entr/issues/32)).

Reply:
in retrospect i should maybe add a comment